moveit_ros/moveit_servo/src/collision_check.cpp

```diff
@@ -91,33 +97,35 @@ planning_scene_monitor::LockedPlanningSceneRO CollisionCheck::getLockedPlanningS
 
 void CollisionCheck::start()
 {
-  timer_ = nh_.createTimer(period_, &CollisionCheck::run, this);
+  timer_ = node_->create_wall_timer(std::chrono::duration<double>(period_), std::bind(&CollisionCheck::run, this));
 }
 
 void CollisionCheck::stop()
 {
-  timer_.stop();
+  timer_->cancel();
 }
 
-void CollisionCheck::run(const ros::TimerEvent& timer_event)
+void CollisionCheck::run()
 {
-  // Log warning when the last loop duration was longer than the period
-  if (timer_event.profile.last_duration.toSec() > period_.toSec())
-  {
-    ROS_WARN_STREAM_THROTTLE_NAMED(ROS_LOG_THROTTLE_PERIOD, LOGNAME,
-                                   "last_duration: " << timer_event.profile.last_duration.toSec() << " ("
-                                                     << period_.toSec() << ")");
-  }
+  // // Log warning when the last loop duration was longer than the period
```

Comment:
Should this be removed? or do we need to fix the timer event

Reply:
Ack, meant to at least leave a TODO in there. I briefly looked around and couldn't find an example or anything indicating we could pass the timer event to the CB.

Am not sure if we should just drop it (and remove the commented code) or try a bit harder to figure that out.

Reply:
I wouldn't mind deleting it. If you do fix it, I think the message could be improved like this:

ROS_WARN_STREAM_THROTTLE_NAMED(ROS_LOG_THROTTLE_PERIOD, LOGNAME, "Collision checking lags desired rate. Actual period: " << timer_event.profile.last_duration.toSec());

Reply:
Deleted it